python/lsst/pipe/base/activator.py

```diff
@@ -0,0 +1,449 @@
+
+            else:
+                log.warn("Not running the task because there is no data to process; "
+                         "you may preview data using \"--show data\"")
+
+
+    def display_tree(self):
+        """
+        This is for WorkflowTask, if a tree structure is present, then it can be outputted in the
+        stdout
+        """
+        if hasattr(self.SuperTask, 'print_tree'):
+            self.SuperTask.print_tree()
+
+    def display_config(self):
+        """
+        Display configuration for new framework, i.e., SuperTask created from scratch and linked
+         together in a WorkFlowTask
+        """
+        if hasattr(self.SuperTask, 'print_config'):
+            self.SuperTask.print_config()
+
+    def generate_dot(self):
+        """
+        It generates a dot file structure of the SuperTask and WorkFlowTask if the methods is
+        defined
+        """
+        if hasattr(self.SuperTask, 'write_tree'):
+            self.SuperTask.write_tree()
+
+
+    @staticmethod
+    def get_target_list(parsed_cmd, **kwargs):
+        """
+        Get the target list from the command  line argument parser, taken from CmdLineTask,
+        getTargetList function.
+
+        :param parsed_cmd: the parsed command object (an argparse.Namespace) returned by
+                           argumentParser.ArgumentParser.parse_args
+                           "ArgumentParser.parse_args".
+        :param kwargs:     any additional keyword arguments. In the default TaskRunner
+                           this is an empty dict, but having it simplifies overriding TaskRunner
+                           for tasks whose run method takes additional arguments
+        :return: A target list
+        """
+        return [(ref, kwargs) for ref in parsed_cmd.id.refList]
+
+
+    @staticmethod
+    def load_super_task(super_taskname):
+        """
+        It search and loads the supertask parsed in the command line. To avoid importing every
+        module on every package defined in TASK_PACKAGES, which introduces and extra overhead and
+        might be of potential conflict, it uses pyclbr which finds classes and functions by looking
+        at the source code of the modules. It is much faster than importing all modules and its
+        much simpler as well. An alternative solution would be to have a predefined register of all
+        task and the modules they live in, or, to have a naming convention where the source file
+        and the Task Class inside have the same name. Currently this is not true for all packages.
+
+        :param super_taskname: The name of the SuperTask
+        :return: the Task and Config instances of the supertask
+        """
+        class_task_instance = None
+        class_config_instance = None
+        super_task_module = None
+
+        for _, package in TASK_PACKAGES.iteritems():  # Loop over all packages
+            for _, module_name, _ in pkgutil.iter_modules(package.__path__):  # Loop over modules
```

Comment:
If you are going to discard the keys then use `for package in TASK_PACKAGES.itervalues():`